packages/ember-runtime/lib/system/core_object.js

```diff
@@ -331,6 +331,75 @@ CoreObject.PrototypeMixin = Mixin.create({
   concatenatedProperties: null,
 
   /**
+    Defines the properties that will be merged from the superclass
+    (instead of overridden).
+
+    By default, when you extend an Ember class a property defined in
+    the subclass overrides a property with the same name that is defined
+    in the superclass. However, there are some cases where it is preferable
+    to build up a property's value by merging the superclass property value
+    with the subclass property's value. An example of this in use within Ember
+    is the `queryParams` property of routes.
+
+    Here is some sample code showing the difference between a merged
+    property and a normal one:
+
+    ```javascript
+    App.BarRoute = Ember.Route.extend({
```

Comment:
IMO, using `Ember.Route` as an example (and query params) is fairly confusing. Something simpler might result in a clearer, shorter example that can be more easily comprehended.